src/services/block.service.ts

```diff
@@ -116,6 +125,96 @@ class BlockService {
       .getRawOne();
     return Number(height);
   }
+
+  async getStatisticsBlocks(
+    offset: number,
+    limit: number,
+    orderBy: keyof BlockEntity,
+    orderDirection: 'DESC' | 'ASC',
+    period: TPeriod,
+  ): Promise<BlockEntity[]> {
+    const query: FindManyOptions<BlockEntity> = {
+      // skip: offset,
+      // take: limit,
+      order: {
+        [orderBy]: orderDirection,
+      },
+    };
+    if (period) {
+      const fromTime = getStartPoint(period) / 1000;
+      query.where = {
+        timestamp: Between(fromTime, new Date().getTime() / 1000),
+      };
+    }
+    if (offset) {
+      query.skip = offset;
+    }
+    if (limit) {
+      query.take = limit;
+    }
+    if (limit) {
+      const statsInfo = await this.getRepository().find(query);
+      return statsInfo;
+    }
+    const count = await this.getRepository().count(query);
+    const take = 500;
+    const skip = Math.round(count / take);
+    let data = [];
+    // get statistics data limit 500 for chart
+    if (count <= take || skip < 2) {
+      const statsInfo = await this.getRepository().find(query);
+      return statsInfo;
+    } else {
+      let index = 0;
+      for (let i = 0; i <= count; i += skip) {
+        const item = await this.getRepository().find({
```

Comment:
Please change this code to not make db calls in a loop. Currently if count is 10000 it looks like db will be requested 10000 times

I suggest to rewrite this using typeorm's query builder and filter blocks using `ROW_NUMBER()` function of sqlite.
Similar to this, I didn't tried this sql, just to show idea:
```sql
SELECT *, row_number() OVER() AS idx
FROM blocks
FILTER (WHERE idx % ${skip} = 0)
```

I know it may be problematic to use query builder of typeorm, and may be tricky to write a query for SQLite, so as a compromise consider fetching all blocks and reducing them on javascript side, it would be much more efficient than performing thousands queries anyway